### PR TITLE
reform job status compare method

### PR DESCRIPTION
**Pull request by shaobo322**
reform job status Compare method, make it human readable like go standard package time.Time

### CODE REVIEW

**Review by steven-zou — APPROVED** [review]
LGTM

**Comment by steven-zou** [comment]
@shaobo322 

Thanks a lot for your contributions. Hope to see more in the future.

File: src/jobservice/period/basic_scheduler.go [hunk 1]
```diff
@@ -155,7 +155,7 @@ func (bs *basicScheduler) UnSchedule(policyID string) error {
 			// Mark job status to stopped to block execution.
 			// The executions here should not be in the final states,
 			// double confirmation: only stop the can-stop ones.
-			if job.RunningStatus.Compare(job.Status(e.Info.Status)) >= 0 {
+			if !job.RunningStatus.Before(job.Status(e.Info.Status)) {
```
Comment (steven-zou) [inline src/jobservice/period/basic_scheduler.go:158]: >=0 means Before() or Equal()?

File: src/pkg/scan/report/base_manager.go [hunk 2]
```diff
@@ -78,7 +78,7 @@ func (bm *basicManager) Create(r *scan.Report) (string, error) {
 
 		// Status conflict
 		if theCopy.StartTime.Add(reportTimeout).After(time.Now()) {
-			if theStatus.Compare(job.RunningStatus) <= 0 {
+			if !theStatus.After(job.RunningStatus) {
```
Comment (steven-zou) [inline src/pkg/scan/report/base_manager.go:81]: <=0 means After() or Equal()?

**Review by steven-zou — CHANGES_REQUESTED** [review]
please do changes

File: src/jobservice/period/basic_scheduler.go (hunk 1 above)
Comment (shaobo322) [reply src/jobservice/period/basic_scheduler.go:158]: image
the old logic is above， so if a.Before(b) is true， means a is earlier than b ( in old logic a.Compare(b) < 0)

File: src/pkg/scan/report/base_manager.go (hunk 2 above)
Comment (shaobo322) [reply src/pkg/scan/report/base_manager.go:81]: the same

File: src/jobservice/period/basic_scheduler.go (hunk 1 above)
Comment (shaobo322) [reply src/jobservice/period/basic_scheduler.go:158]: you mean I should write like this?
```
job.RunningStatus.After(job.Status(e.Info.Status)) || job.RunningStatus.Equal(job.Status(e.Info.Status))
```

Comment (steven-zou) [reply src/jobservice/period/basic_scheduler.go:158]: Yup! >=0 means your `Before` and `Equal `

<=0 means your `After` and `Equal`.

@shaobo322

**Comment by steven-zou** [comment]
@shaobo322

Fix conflicts and do changes? Then we can merge.

**Comment by shaobo322** [comment]
> @shaobo322
> 
> Fix conflicts and do changes? Then we can merge.

done @steven-zou

File: src/jobservice/period/basic_scheduler.go [hunk 3]
```diff
@@ -145,7 +145,7 @@ func (bs *basicScheduler) UnSchedule(policyID string) error {
 			// Mark job status to stopped to block execution.
 			// The executions here should not be in the final states,
 			// double confirmation: only stop the can-stop ones.
-			if job.RunningStatus.Compare(job.Status(e.Info.Status)) >= 0 {
+			if job.RunningStatus.After(job.Status(e.Info.Status)) || job.RunningStatus.Equal(job.Status(e.Info.Status)) {
```
Comment (steven-zou) [inline src/jobservice/period/basic_scheduler.go:148]: actually, a better way is :

`job.Status(e.Info.Status.Before(job.RunningStatus) || job.Status(e.Info.Status).Equal(job.RunningStatus)`

anyway, the current one is ok

**Review by steven-zou — APPROVED** [review]
LGTM

**Review by heww — APPROVED** [review]

**Pull request merged**